### PR TITLE
Cross build for sbt 1.0

### LINKED ISSUE #198: Support SBT 1.0
SBT 1.0.0-M5 was released. It would be nice if sbt-release supported that version of sbt.

### PR DESCRIPTION
**Pull request by leonardehrenfried**
Fixes #198.

I took a stab at implementing sbt 1.0 cross building but I have hit a road block.

The compatibility class appears to not be found when I load the plugin. 

```
/home/travis/build/sbt/sbt-release/src/main/scala/ReleaseExtra.scala:19: not found: value Compat
val (newSt, result) = Compat.runTaskAggregated(releaseSnapshotDependencies in thisRef, st)                         
```
Is it possible that that has something to do with the fact that the plugin requires itself? Maybe during the recursive loading the `/src/scala-sbt-0.13` path is ignored?

/cc @dwijnand @xuwei-k

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
Actually, I've managed to resolve this issue by depending on the JAR itself, not a source dependency.

**Comment by leonardehrenfried** [comment]
The test for 0.13 fail because of the following error:

```
java.lang.RuntimeException: Aborting release: untracked files. Remove them or specify 'releaseIgnoreUntrackedFiles := true' in settings
```
Was this the case before these changes?

**Comment by leonardehrenfried** [comment]
Also, `sbt.internal.Load` became private in 1.0. Is there a replacement for it?

File: build.sbt [hunk 1]
```diff
@@ -6,18 +6,39 @@ name := "sbt-release"
...
+libraryDependencies ++= Seq("org.specs2" %% "specs2-core" % "3.9.4" % "test")
 resolvers += "scalaz-bintray" at "http://dl.bintray.com/scalaz/releases"
 
 // Scripted
-scriptedSettings
-scriptedLaunchOpts <<= (scriptedLaunchOpts, version) { case (s,v) => s ++
-  Seq("-Xmx1024M", "-XX:MaxPermSize=256M", "-Dplugin.version=" + v)
+
+// workaround from https://github.com/sbt/sbt/issues/3325
+scriptedSettings.filterNot(_.key.key.label == libraryDependencies.key.label)
```
Comment (xuwei-k) [inline build.sbt]: This workaround no longer necessary. sbt 0.13.16-RC1 released

File: README.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ This sbt plugin provides a customizable release process that you can add to your
 
 **Notice:** This README contains information for the latest release. Please refer to the documents for a specific version by looking up the respective [tag](https://github.com/sbt/sbt-release/tags).
 
+[![Build Status](https://travis-ci.org/sbt/sbt-release.svg?branch=master)](https://travis-ci.org/leonardehrenfried/sbt-release)
```
Comment (xuwei-k) [inline README.md]: s/leonardehrenfried/sbt

File: build.sbt [hunk 3]
```diff
@@ -6,18 +6,19 @@ name := "sbt-release"
 homepage := Some(url("https://github.com/sbt/sbt-release"))
 licenses := Seq("Apache-2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0"))
 
+crossSbtVersions := Vector("0.13.15", "1.0.0-M6")
```
Comment (xuwei-k) [inline build.sbt]: 1.0.0-RC2

**Comment by dwijnand** [comment]
@leonardehrenfried Currently we have https://github.com/sbt/sbt/issues/3296 to track the need for APIs that have been made private in sbt 1.

**Comment by leonardehrenfried** [comment]
Does anyone have a tip on how to translate this construct to 1.0? https://github.com/sbt/sbt-release/pull/207/commits/a23500f151ccc793ea9b6b84915e68783a8ad119#diff-cd0ff4ae2e1a0625229c3763f4a388d9R48

**Comment by leonardehrenfried** [comment]
@xuwei-k @dwijnand: This plugin now compiles for 0.13 and 1.0. I'm down to two test failures and I need your help with this.

Testing the sbt-doge integration fails because sbt-doge is not available for sbt 1.0. Is it possible to ignore this scripted test until an updated version is published?

The tag test fails because of the version is extracted incorrectly: https://travis-ci.org/leonardehrenfried/sbt-release/jobs/257069070#L897
I'm assuming it is due to this change: https://github.com/sbt/sbt-release/pull/207/commits/6775ebde6c8803dfa0f9a14ee33adfb60f65b9c1

How do I extract it correctly?

Also, I had to modify the tests because in some sbt version between 0.13.8 and 0.13.16-RC1 sbt started to automatically modify `project/build.properties` which lead to aborted tests due to untracked files.

Please review.

**Comment by xuwei-k** [comment]
>  sbt-doge is not available for sbt 1.0.

https://github.com/sbt/sbt/pull/2613

**Comment by leonardehrenfried** [comment]
Thanks for the link.

I'm down to a single test failing in 1.0 because CrossPerProjectPlugin is missing. I tried to put a dummy plugin in `Compat`. But that doesn't seem to work.

Any ideas?

**Comment by leonardehrenfried** [comment]
It's bit frustrating because I'm very, very close to getting everything to go green. It's the single doge integration test that is failing.

Is there a way to ignore a scripted test for a certain version?

**Comment by leonardehrenfried** [comment]
I don't think I can get the SBT doge test working in 1.0.

I'm going to change strategies and create a subproject for each the 0.13
and 1.0 tests and only run them for their respective versions.

Awkward, but, as Dale already said, it's unlikely that any core plugin can
become 1.0-only any time soon.

**Comment by xuwei-k** [comment]
I have added some commits and squash merged in https://github.com/sbt/sbt-release/commit/eaa51526d69258f94ae3c72d822ca7e0ae5b1a1c
Thanks for contributions!

**Pull request closed without merging**